packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/method_channel/method_channel_google_maps_flutter.dart

```diff
@@ -519,44 +516,7 @@ class MethodChannelGoogleMapsFlutter extends GoogleMapsFlutterPlatform {
           return controller;
         },
       );
-    }
-    return buildView(
-      creationId,
-      onPlatformViewCreated,
-      initialCameraPosition: initialCameraPosition,
-      markers: markers,
-      polygons: polygons,
-      polylines: polylines,
-      circles: circles,
-      tileOverlays: tileOverlays,
-      gestureRecognizers: gestureRecognizers,
-      mapOptions: mapOptions,
-    );
-  }
-
-  @override
-  Widget buildView(
-    int creationId,
-    PlatformViewCreatedCallback onPlatformViewCreated, {
-    required CameraPosition initialCameraPosition,
-    Set<Marker> markers = const <Marker>{},
-    Set<Polygon> polygons = const <Polygon>{},
-    Set<Polyline> polylines = const <Polyline>{},
-    Set<Circle> circles = const <Circle>{},
-    Set<TileOverlay> tileOverlays = const <TileOverlay>{},
-    Set<Factory<OneSequenceGestureRecognizer>>? gestureRecognizers,
-    Map<String, dynamic> mapOptions = const <String, dynamic>{},
-  }) {
-    final Map<String, dynamic> creationParams = <String, dynamic>{
-      'initialCameraPosition': initialCameraPosition.toMap(),
-      'options': mapOptions,
-      'markersToAdd': serializeMarkerSet(markers),
-      'polygonsToAdd': serializePolygonSet(polygons),
-      'polylinesToAdd': serializePolylineSet(polylines),
-      'circlesToAdd': serializeCircleSet(circles),
-      'tileOverlaysToAdd': serializeTileOverlaySet(tileOverlays),
-    };
-    if (defaultTargetPlatform == TargetPlatform.android) {
+    } else if (defaultTargetPlatform == TargetPlatform.android) {
```

Comment:
With the new combined structure it would be clearer to do

```dart
if (defaultTargetPlatform == TargetPlatform.android) {
  if (useAndroidViewSurface) {
    ...
  } else {
    ...
  }
} else if ...
```

instead of 

```dart
if (defaultTargetPlatform == TargetPlatform.android &&
        useAndroidViewSurface) {
  ...
} else if (defaultTargetPlatform == TargetPlatform.android) {
  ...
} else if ...
```